packages/winpanda/extra/src/winpanda/tests/unit/test_common_utils.py

```diff
@@ -0,0 +1,140 @@
+    """Unpack wrong tarfile path."""
+    with pytest.raises(ValueError):
+        utils.unpack('', '')
+
+
+@mock.patch('tarfile.open')
+@mock.patch('os.path.exists', return_value=True)
+def test_unpack_should_return_destination_location(mock_tar, *args):
+    """Check is unpack file result provide valid location."""
+    mock_tar.__enter__.return_value = mock.Mock()
+    location = utils.unpack('', 'fake_path')
+    # Check only end of string because of absolute path
+    assert location.endswith('fake_path')
+
+
+@mock.patch('common.utils.Path.exists', return_value=False)
+def test_rm_nonexistent_dir_should_fail(*args):
+    """Remove empty path without exceptions and None result."""
+    assert utils.rmdir('path') is None
+
+
+@mock.patch('common.utils.Path.exists', return_value=True)
+@mock.patch('common.utils.Path.is_symlink', return_value=True)
+def test_rm_symlink_should_fail(*args):
+    """Try remove symlink and raise exception."""
+    with pytest.raises(OSError):
+        utils.rmdir('/tmp/.000')
+
+
+@mock.patch('common.utils.Path.exists', return_value=True)
+@mock.patch('common.utils.Path.is_symlink', return_value=False)
+@mock.patch('common.utils.Path.is_reserved', return_value=True)
+def test_rm_reserved_should_fail(*args):
+    """Try remove reserved path and raise exception."""
+    with pytest.raises(OSError):
+        utils.rmdir('/tmp/.000')
+
+
+@mock.patch('common.utils.Path.exists', return_value=True)
+@mock.patch('common.utils.Path.is_symlink', return_value=False)
+@mock.patch('common.utils.Path.is_reserved', return_value=False)
+@mock.patch('common.utils.Path.is_dir', return_value=False)
+def test_rm_not_dir_should_fail(*args):
+    """Try remove file and raise exception."""
+    with pytest.raises(OSError):
+        utils.rmdir('/tmp/.000')
+
+
+def test_run_external_command_process_error_should_fail():
+    """Check external command execution CalledProcessError handling."""
+    def run(*args, **kwargs):
+        # TODO need to be checked on Exception without output and stderr params
+        raise CalledProcessError(1, 'command', 'out', 'err')
+
+    with mock.patch('subprocess.run', side_effect=run):
+        with pytest.raises(ExternalCommandError):
+            utils.run_external_command('')
+
+
+@mock.patch('subprocess.run', side_effect=SubprocessError)
+def test_run_external_command_subprocess_error_should_fail(*args):
+    """Check external command execution SubprocessError handling."""
+    with pytest.raises(ExternalCommandError):
+        utils.run_external_command('')
+
+
+@mock.patch('subprocess.run', side_effect=OSError)
+def test_run_external_command_os_error_should_fail(*args):
+    """Check external command execution OSError handling."""
+    with pytest.raises(ExternalCommandError):
+        utils.run_external_command('')
+
+
+def test_successful_run_external_command_should_provide_output():
+    """Check do we have external command execution output in function result."""
```

Comment:
`subprocess.run` returns a `CompletedProcess` object, so that is what `run_external_command` returns. This test seems to suggest that it will return the process output.

Reply:
I have change assertion flow, now it is checking execution result and run input data.